lib/miq_automation_engine/service_models/miq_ae_service_generic_object.rb

```diff
@@ -4,6 +4,14 @@ class MiqAeServiceGenericObject < MiqAeServiceModelBase
     include MiqAeServiceRemoveFromVmdb
     require 'drb'
 
+    def initialize(*args)
+      super(*args)
+
+      @object.property_methods.each do |method_name|
+        define_ae_method_from_generic_object(method_name)
+      end
```

Comment:
All this logic should be pushed into a method.  The method may loop over the method names and call another method to define the actual methods for this object, but I would expect a single, clear method name here.

Also, what about the `property_attributes`?  They need to be exposed as well.